packages/functions/src/reports/BookedDonationsReport.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  BookedDonationWithDonor,
+  DbAppointment,
+  FunctionsApi,
+  Hospital,
+} from "@zm-blood-components/common";
+import { validateAppointmentEditPermissions } from "../coordinator/UserValidator";
+import { getAppointmentsByHospital } from "../dal/AppointmentDataAccessLayer";
+import { getDonors } from "../dal/DonorDataAccessLayer";
+
+export default async function (
+  request: FunctionsApi.GetBookedDonationsInHospitalRequest,
+  callerId: string
+): Promise<FunctionsApi.GetBookedDonationsInHospitalResponse> {
+  await validateAppointmentEditPermissions(
```

Comment:
This is not enough.
See in files like `GetDonorsHandler` how we have switch-case on the admin role and decide which roles can run this request. 

I think you should also ask Adi who has permissions for this API

Reply:
Currently undefined, 
ZM coordinators are currently the only relevant group.
Marked //TODO